In [1]:
import pandas as pd
import numpy as np

In [8]:
company_list = pd.read_csv('../../file/subject/chapter3/ZGS_info.csv',names=['维度','经度','经营范围','经营范围2','地址',
                                                                             '代码','高管','名称','上市公司名称','状态','注册资本',
                                                                            '上市公司投资比例'])

### 1.检查数据是否完整

#### 1.1检查长度

In [11]:
len(company_list)

56747

#### 1.2聚合

In [13]:
company_list.groupby(by='上市公司名称').count()

维度  经度  经营范围  经营范围2  地址  代码  高管  名称  状态  注册资本  上市公司投资比例
上市公司名称                                                                
TCL集团股份有限公司    72  72    74     74  74  74  74  74  74    73        74
一品红药业股份有限公司     3   3     3      3   3   3   3   3   3     3         3
一汽轿车股份有限公司      7   7     7      7   7   7   7   7   7     7         7
万华化学集团股份有限公司   21  21    22     22  22  22  22  22  22    22        22
万向德农股份有限公司      1   1     1      1   1   1   1   1   1     1         1
...            ..  ..   ...    ...  ..  ..  ..  ..  ..   ...       ...
齐峰新材料股份有限公司     2   2     2      2   2   2   2   2   2     2         2
龙元建设集团股份有限公司   68  68    75     75  76  76  76  76  74    72        76
龙建路桥股份有限公司     43  43    45     45  45  45  45  45  45    44        45
龙星化工股份有限公司      5   5     5      5   5   5   5   5   5     5         5
龙蟒佰利联集团股份有限公司   6   6     6      6   6   6   6   6   6     6         6

[3325 rows x 11 columns]

#### 1.3结论
3325条数据,比原数据3534少了，对比是因为无法识别中文字符，修改后重新爬取

### 2.合并两次爬取的数据

#### 2.1读取新爬取的数据，注意列名要相同

In [24]:
company_list_add =  pd.read_csv('../../file/subject/chapter3/ZGS_info_add_crawl.csv',
                                names=['维度','经度','经营范围','经营范围2','地址',
                                       '代码','高管','名称','上市公司名称','状态',
                                       '注册资本','上市公司投资比例'])

In [26]:
#注意concat的对象是一个列表
data = pd.concat([company_list,company_list_add])

In [28]:
data.to_excel('../../file/subject/chapter3/company_all.xlsx',sheet_name='data')